wordpress.org/public_html/wp-content/plugins/theme-directory/rest-api/class-theme-review-stats.php

```diff
@@ -0,0 +1,344 @@
+
+use WP_Error;
+use WP_REST_Controller, WP_REST_Server, WP_REST_Response;
+
+defined( 'WPINC' ) || die();
+
+/**
+ *
+ * @see WP_REST_Controller
+ */
+class Theme_Review_Stats extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'themes/v1';
+		$this->rest_base = 'stats';
+
+		$this->register_routes();
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_uploaded_themes' ),
+				'permission_callback' => array( $this, 'get_items_permissions' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/byThemeType',
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_by_theme_type' ),
+				'permission_callback' => array( $this, 'get_items_permissions' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/bySegment',
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_by_segment' ),
+				'permission_callback' => array( $this, 'get_items_permissions' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/byAuthorType',
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_by_author_type' ),
+				'permission_callback' => array( $this, 'get_items_permissions' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/reviewDays',
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_review_days' ),
+				'permission_callback' => array( $this, 'get_items_permissions' ),
+			)
+		);
+	}
+
+	/**
+	 * Get start date param from request.
+	 *
+	 * @returns string
+	 */
+	function get_start_date( $request ) {
+		$start_date = $request->get_param( 'startDate' );
+
+		if ( empty( $start_date ) ) {
+			$start_date = date( 'Y-m-d', strtotime( '-3 months' ) ); // Default to 3 months ago
+		}
+
+		return $start_date;
+	}
+
+	/**
+	 * Return whether the user can retrieve the data.
+	 *
+	 * @return bool
+	 */
+	function get_items_permissions( $request ) {
+		return current_user_can( 'edit_posts' );
+	}
+
+	/**
+	 * Returns an array of elements after the start date.
+	 */
+	function filter_items_before( $arr, $start_date ) {
+		return array_filter(
+			$arr,
+			function ( $row ) use ( $start_date ) {
+				return strtotime( $row[0] ) >= strtotime( $start_date );
+			}
+		);
+	}
+
+	/**
+	 * Returns a list averages review times by month.
+	 *
+	 * @return void
+	 */
+	public function get_average_review_days( $start_date ) {
```

Comment:
Should this function have some kind of caching given everything is ultimately pulled from here? Even if it's only an hour.